Files/Views/ModernShellPage.xaml.cs

```diff
@@ -103,21 +555,21 @@ protected override void OnNavigatedTo(NavigationEventArgs eventArgs)
             NavParams = eventArgs.Parameter.ToString();
         }
 
-        private ItemViewModel viewModel = null;
-        private Interaction interactionOperation = null;
-        private readonly CurrentInstanceViewModel instanceViewModel = new CurrentInstanceViewModel();
-
+        AppServiceConnection Connection = null;
         private void Page_Loaded(object sender, RoutedEventArgs e)
         {
-            viewModel = new ItemViewModel();
-            interactionOperation = new Interaction();
-
+            Connection = new AppServiceConnection();
+            FilesystemViewModel = new ItemViewModel(this, ref Connection);
+            FilesystemViewModel.InitializeAppServiceConnection();
+            InteractionOperations = new Interaction(this, ref Connection);
```

Comment:
Should the AppInstance be passed by reference here also?

Reply:
Pass by reference if you want the reference change to apply in other places. Technically already everything is a reference but changing it will create a new reference without the ref keyword.

Reply:
Gave92's recent commits make this question obsolete.